components/org.wso2.carbon.identity.login.resolver/org.wso2.carbon.identity.login.resolver.regex/src/main/java/org/wso2/carbon/identity/login/resolver/regex/RegexLoginResolver.java

```diff
@@ -0,0 +1,205 @@
+            UniqueIDUserStoreManager userStoreManager = UserResolverUtil.getUserStoreManager(tenantDomain);
+            claimManager = userRealm.getClaimManager();
+            for (String claimURI : allowedAttributes) {
+                Claim claim = claimManager.getClaim(claimURI);
+                if (claim == null) {
+                    continue;
+                }
+                String regex = claim.getRegEx();
+                if (StringUtils.isBlank(regex)) {
+                    continue;
+                }
+                Pattern pattern = Pattern.compile(regex);
+                if (pattern.matcher(loginIdentifier).matches()) {
+                    authenticationResult = userStoreManager.
+                            authenticateWithID(claimURI, loginIdentifier, credential, StringUtils.EMPTY);
+                    if (AuthenticationResult.AuthenticationStatus.SUCCESS.
+                            equals(authenticationResult.getAuthenticationStatus())) {
+                        break;
+                    }
+                }
+            }
+            /*
+            If allowed attributes has only username claim, get authenticationResult even if
+            the username claim has no configured regex pattern.
+             */
+            if (allowedAttributes.size() == 1 && allowedAttributes.contains(
+                    UserCoreClaimConstants.USERNAME_CLAIM_URI)) {
+                authenticationResult =
+                        userStoreManager.authenticateWithID(UserCoreClaimConstants.USERNAME_CLAIM_URI, loginIdentifier,
+                                credential, StringUtils.EMPTY);
+            }
+        } catch (UserStoreException e) {
+            log.error("An error occurred while resolving authentication result.", e);
+        }
+        return authenticationResult;
+    }
+
+    /**
+     * Sets the resolved user result object with the user details if a valid user is found.
+     *
+     * @param userStoreManager   The user store manager.
+     * @param claimURI           The claim URI which should be used to extract the user.
+     * @param loginIdentifier    The login identifier provided by the user.
+     * @param resolvedUserResult The dummy ResolvedUserResult object for which the user details should be added to.
+     * @param claim              The claim object.
+     * @throws org.wso2.carbon.user.core.UserStoreException If there is an error while extracting a list of users with
+     *                                                      the ID.
+     */
+    private void setResolvedUserResult(UniqueIDUserStoreManager userStoreManager, String claimURI,
+                                       String loginIdentifier, ResolvedUserResult resolvedUserResult, Claim claim)
+            throws org.wso2.carbon.user.core.UserStoreException {
+
+        List<User> userList = userStoreManager.getUserListWithID(claimURI, loginIdentifier, null);
+        if (userList.size() == 1) {
```

Comment:
Did you check whether we are getting always a list object ? what happens if the list becomes null due to some reason ? if it is returning a not null object then this is fine.

Reply:
The method seems to be returning a List object all the time. However, I will add a null check just in case if it returns null.

Reply:
Fixed with https://github.com/wso2-extensions/identity-governance/pull/652/commits/8850e6ed66ee47b980c57d32cb7918c9dde6d577.